base/database/testing.go

```diff
@@ -36,7 +37,7 @@ func CheckCachesValid() (bool, error) {
 	valid := true
 	var aad []models.AdvisoryAccountData
 
-	tx := Db.Begin()
+	tx := Db.BeginTx(base.Context, nil)
```

Comment:
We're also using context in kafka reader & writer implementation & API calls. It'd be nice to use drerived contexts there, which would allow us to stop all processing when receiving a signal.

Reply:
Ok, good idea.